src/main/java/seedu/algobase/logic/parser/AddCommandParser.java

```diff
@@ -38,9 +38,9 @@
     public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_AUTHOR, PREFIX_WEBLINK, PREFIX_DESCRIPTION,
-                        PREFIX_TAG);
+                        PREFIX_TAG, PREFIX_DIFFICULTY, PREFIX_REMARK, PREFIX_SOURCE);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_DESCRIPTION, PREFIX_AUTHOR, PREFIX_WEBLINK)
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME)
```

Comment:
While removing the checks for the prefixes here will stop the method from throwing a `ParseException`, the program will encounter an error in lines 48 to 55 if not all the prefixes are specified.

For example, if the user does not specify the `PREFIX_AUTHOR`, the program will encounter an error in line 49 when it tries to run `argMultimap.getValue(PREFIX_AUTHOR)`.

For each of the optional prefixes, it would be good to first check if the prefix exists (i.e. `argMultimap.getValue(PREFIX_AUTHOR).isPresent()`). If the prefix is not present, then we should set it to be the default value.